This patch undoes parts of the following upstream patch:

commit 19442df2cd9a5bc8ec4deded7046ea7aca1d50a2
Author: Kota Tsuyuzaki <tsuyuzaki.kota@lab.ntt.co.jp>
Date:   Mon Feb 20 14:12:43 2017 -0800

    Fix warning on automake tool

    ..........

    Change-Id: I0e79bed7755a1f286b746a70fcf56fdc972bfd5d

diff -u liberasurecode-1.5.0/Makefile.am liberasurecode-1.5.0-p3/Makefile.am
--- liberasurecode-1.5.0/Makefile.am	2017-06-29 11:36:37.000000000 -0600
+++ liberasurecode-1.5.0-p3/Makefile.am	2017-07-05 20:44:59.041285796 -0600
@@ -46,6 +46,11 @@
 	   $(LN_S) liberasurecode/config_liberasurecode.h)
 
 test: check
+	$(eval SONAMES := $(shell find $(abs_top_builddir) -name '*.so'))
+	$(eval SODIRS := $(dir $(SONAMES)))
+	$(eval LD_LIBRARY_PATH := LD_LIBRARY_PATH="$(LD_LIBRARY_PATH):$(subst / ,/:,$(SODIRS))")
+	$(eval DYLD_LIBRARY_PATH := DYLD_LIBRARY_PATH="$(DYLD_LIBRARY_PATH):$(subst / ,/:,$(SODIRS))")
+	$(eval DYLD_FALLBACK_LIBRARY_PATH := DYLD_FALLBACK_LIBRARY_PATH=$(DYLD_FALLBACK_LIBRARY_PATH):"$(subst / ,/:,$(SODIRS))")
 	@$(LD_LIBRARY_PATH) $(DYLD_LIBRARY_PATH) $(DYLD_FALLBACK_LIBRARY_PATH) \
 		./test/liberasurecode_test
 	@$(LD_LIBRARY_PATH) $(DYLD_LIBRARY_PATH) $(DYLD_FALLBACK_LIBRARY_PATH) \
@@ -60,6 +65,11 @@
 	--malloc-fill=A5 --free-fill=DE --fullpath-after=. --trace-children=yes
 
 valgrind-test: check
+	$(eval SONAMES := $(shell find $(abs_top_builddir) -name '*.so'))
+	$(eval SODIRS := $(dir $(SONAMES)))
+	$(eval LD_LIBRARY_PATH := LD_LIBRARY_PATH="$(LD_LIBRARY_PATH):$(subst / ,/:,$(SODIRS))")
+	$(eval DYLD_LIBRARY_PATH := DYLD_LIBRARY_PATH="$(DYLD_LIBRARY_PATH):$(subst / ,/:,$(SODIRS))")
+	$(eval DYLD_FALLBACK_LIBRARY_PATH := DYLD_FALLBACK_LIBRARY_PATH=$(DYLD_FALLBACK_LIBRARY_PATH):"$(subst / ,/:,$(SODIRS))")
 	@$(LD_LIBRARY_PATH) $(DYLD_LIBRARY_PATH) $(DYLD_FALLBACK_LIBRARY_PATH) $(VALGRIND_EXEC_COMMAND) \
 		./test/alg_sig_test
 	@$(LD_LIBRARY_PATH) $(DYLD_LIBRARY_PATH) $(DYLD_FALLBACK_LIBRARY_PATH) $(VALGRIND_EXEC_COMMAND) \
